sdk/v2/core/validations.go

```diff
@@ -0,0 +1,35 @@
+package core
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var projectIDRegex = regexp.MustCompile(`^[a-z][a-z\d-]*[a-z\d]$`)
+
+// nolint: lll
+var gitCloneURLRegex = regexp.MustCompile(`^(?:(?:https?://)|(?:git@))[\w:/\-\.\?=@&]+\.git$`)
+
+// ValidateProjectID checks if a given id is valid.
+func ValidateProjectID(id string) error {
+	idMatch := projectIDRegex.MatchString(id)
+	if !idMatch || len(id) < 3 || len(id) > 63 {
+		return fmt.Errorf("invalid value %q for project id"+
+			" (3-63 characters, first char must be"+
+			" a letter, lowercase letters only)", id)
```

Comment:
Nit: To enforce a consistent style, I'd prefer this were formatted like this:

```
return fmt.Errorf(
	"invalid value %q for project id"+
		" (3-63 characters, first char must be"+
		" a letter, lowercase letters only)",
	id,
)
```

The rationale for this is twofold.

1. Once a line gets so long that we need to break it up across a few lines, our brains swap from horizontal mode to vertical mode. Instead of looking across to see comma delimited args, we look down to see line break delimited args. As such, we should try to keep one arg per line because it makes everything more clear to someone who's now scanning down. Don't make people scan down _and_ across. idk if that makes sense.

2. If the closing parenthesis is on its own outdented line, it's visually much more obvious what opening paren it corresponds to.